crates/router/src/core/payments/helpers.rs

```diff
@@ -5406,6 +5413,539 @@ impl ApplePayData {
+        signed_data.append(&mut recipient_id.as_bytes().to_vec());
+        signed_data.append(&mut get_little_endian_format(length_of_protocol_version));
+        signed_data.append(&mut protocol_version.as_bytes().to_vec());
+        signed_data.append(&mut get_little_endian_format(length_of_signed_key));
+        signed_data.append(&mut signed_key.as_bytes().to_vec());
+
+        Ok(signed_data)
+    }
+
+    // Derive a shared key using ECDH
+    fn get_shared_key(
+        &self,
+        ephemeral_public_key_bytes: &[u8],
+    ) -> CustomResult<Vec<u8>, errors::GooglePayDecryptionError> {
+        let group = openssl::ec::EcGroup::from_curve_name(openssl::nid::Nid::X9_62_PRIME256V1)
+            .change_context(errors::GooglePayDecryptionError::DerivingEcGroupFailed)?;
+
+        let mut big_num_context = openssl::bn::BigNumContext::new()
+            .change_context(errors::GooglePayDecryptionError::BigNumAllocationFailed)?;
+
+        let ec_key = openssl::ec::EcPoint::from_bytes(
+            &group,
+            ephemeral_public_key_bytes,
+            &mut big_num_context,
+        )
+        .change_context(errors::GooglePayDecryptionError::DerivingEcKeyFailed)?;
+
+        // create an ephemeral public key from the given bytes
+        let ephemeral_public_key = openssl::ec::EcKey::from_public_key(&group, &ec_key)
+            .change_context(errors::GooglePayDecryptionError::DerivingPublicKeyFailed)?;
+
+        // wrap the public key in a PKey
+        let ephemeral_pkey = PKey::from_ec_key(ephemeral_public_key)
+            .change_context(errors::GooglePayDecryptionError::DerivingPublicKeyFailed)?;
+
+        // perform ECDH to derive the shared key
+        let mut deriver = Deriver::new(&self.private_key)
+            .change_context(errors::GooglePayDecryptionError::DerivingSharedSecretKeyFailed)?;
+
+        deriver
+            .set_peer(&ephemeral_pkey)
+            .change_context(errors::GooglePayDecryptionError::DerivingSharedSecretKeyFailed)?;
+
+        let shared_key = deriver
+            .derive_to_vec()
+            .change_context(errors::GooglePayDecryptionError::DerivingSharedSecretKeyFailed)?;
+
+        Ok(shared_key)
+    }
+
+    // Derive symmetric key and MAC key using HKDF
+    fn derive_key(
+        &self,
+        ephemeral_public_key_bytes: &[u8],
+        shared_key: &[u8],
+    ) -> CustomResult<Vec<u8>, errors::GooglePayDecryptionError> {
+        // concatenate ephemeral public key and shared key
+        let input_key_material = [ephemeral_public_key_bytes, shared_key].concat();
+
+        // initialize HKDF with SHA-256 as the hash function
+        // Salt is not provided as per the Google Pay documentation
+        // https://developers.google.com/pay/api/android/guides/resources/payment-data-cryptography#encrypt-spec
+        let hkdf: ::hkdf::Hkdf<sha2::Sha256> = ::hkdf::Hkdf::new(None, &input_key_material); // 32 zeroed bytes as salt
```

Comment:
```suggestion
        let hkdf: ::hkdf::Hkdf<sha2::Sha256> = ::hkdf::Hkdf::new(None, &input_key_material);
```

Can remove the comment, since we pass `None` instead.